src/status_im/chat/models/delete_message_for_me_test.cljs

```diff
@@ -2,114 +2,118 @@
   (:require [cljs.test :refer-macros [deftest is testing]]
             [status-im.chat.models.delete-message-for-me :as
              delete-message-for-me]
+            [status-im.chat.models.message-list :as message-list]
             [status-im.utils.datetime :as datetime]))
 
 (defonce mid "message-id")
 (defonce cid "chat-id")
 
 (deftest delete-for-me
-  (let [db      {:messages {cid {mid {:id mid}}}}
-        message {:message-id mid :chat-id cid}]
-    (testing "delete for me"
-      (let [expected  {:db {:messages {"chat-id" {"message-id"
-                                                  {:id              "message-id"
-                                                   :deleted-for-me? true}}}}
-                       :utils/dispatch-later
-                       [{:dispatch [:chat.ui/delete-message-for-me-and-sync
-                                    {:chat-id "chat-id" :message-id "message-id"}]
-                         :ms       1000}]}
-            result    (delete-message-for-me/delete {:db db} message 1000)
-            timestamp (+ (datetime/timestamp) 1000)]
-        (is (= (update-in result [:db :messages "chat-id" "message-id"] dissoc :deleted-for-me-undoable-till)
-               expected))
-        (is (-> (get-in result [:db :messages "chat-id" "message-id" :deleted-for-me-undoable-till])
-                (- timestamp)
-                js/Math.abs
-                (< 10)))))
-    (testing "should return nil if message in db"
-      (is (= (delete-message-for-me/delete {:db {:messages []}} message 1000)
-             nil)))))
+  (with-redefs [message-list/rebuild-message-list identity]
```

Comment:
mock `rebuild-message-list` in test

Reply:
@yqrashawn, of course there are exceptions, but it's usually considered a better practice to not mock the code that you own and control because you end up with more brittle tests. This is especially true in Clojure, where devs often only mock functions with side effects. In this example, `rebuild-message-list` is a simple, pure function relying on what's inside the app db, so I believe you could change the test to not redefine a pure function.

There are other instances of this problem in the PR, but I'm leaving only this single comment.